feat: New connection management docs (#29166)
supabase/supabase: apps/docs/content/guides/platform/ipv4-address.mdx

subtitle: 'Attach an IPv4 address to your database'
---

> The Supabase IPv4 add-on provides a dedicated IPv4 address for your Postgres database connection. It costs $0.0055 per hour enabled, or about $4 per month if used continuously. It can be configured in the [Add-Ons Settings](https://supabase.com/dashboard/project/_/settings/addons).

Reviewer: ```suggestion
The Supabase IPv4 add-on provides a dedicated IPv4 address for your Postgres database connection. It can be configured in the [Add-Ons Settings](/dashboard/project/_/settings/addons).
```

Let's not put pricing information in the docs (difficult to keep in sync)

Author: The IPv4 Add-Ons page had the pricing beforehand and if we remove the value, the only reference to its price will be directly in the Dashboard. 

I'm not opposed to taking out the information, but outside of our [changelogs](https://github.com/orgs/supabase/discussions/28438), we don't mention anywhere else that IPv4 is charged for by the hour.

It's important that it is somewhere public because using direct connections is a strategy developers can use to debug connection errors. They'll be much less likely to experiment with it if they think its a flat rate of $4.

Reviewer: good context, I think we should just update the text at the source so we only have one place to correct it in future

made a PR here to merge in this one https://github.com/supabase/supabase/pull/29210, just needs frontend and kevin to look at it because dashboard/billing

## Understanding IP addresses

The Internet Protocol (IP) addresses devices on the internet. There are two main versions:

- **IPv4**: The older version, with a limited address space.
- **IPv6**: The newer version, offering a much larger address space and the future-proof option.

## When you need the IPv4 add-on:

- When using the direct connection string in an IPv6-incompatible network instead of Supavisor or client libraries.
- When you need a dedicated IP address for your direct connection string

## Enabling the IPv4 add-on

> You can enable the IPv4 add-on in your project add-on settings page. Briefly disabling the database might occur during the process.

Reviewer: ```suggestion
You can enable the IPv4 add-on in your project add-on settings page. Your database might be briefly disabled to add the add-on.
```

Approximately how long is briefly?

Author: > Approximately how long is briefly?

Generally 2 minutes

## Read replicas and IPv4 add-on

> When using the add-on, each database (including read replicas) receives an IPv4 address with an additional $4 charge per replica.

Reviewer: ```suggestion
When using the add-on, each database (including read replicas) receives an IPv4 address. Each replica adds to the total IPv4 cost.
```
